### ISSUE TITLE
Direct7toDirect11wrapper linux

### ISSUE THREAD

**Issue by Smokolak** [Kirides/GD3D11#14]
Hi, this is more a question, is it possible to run this under Wine? What packages I need to register in it to work? Is it even possible?

**Comment by neVERberleRfellerER** [comment]
Yes it is. It works well with DXVK+native d3dxXX (as specified in DXVK README) +native dsound (it seemingly work with builtin libs, but music sometimes does not work and some sound effects cause crashes). The only problem encountered by me is https://bugs.winehq.org/show_bug.cgi?id=48483 - and it's annoying enought to not use D3D11 renderer with Wine.

**Comment by Kirides** [comment]
>This issue is caused by forced borderless fullscreen mode. One way how to replicate is by setting
> zStartupWindowed=1
> with systempack and then cursor is always visible even without GD3D11.

Can someone try this without windowed mode?

**Comment by neVERberleRfellerER** [comment]
It used to be the same with `zStartupWindowed=0`. This problem somehow vanished, but I don't know what fixed it, but my guess is that something in wine-staging did. I didn't have chance to test vanilla wine yet, because last time I tried I get crash at startup.

**Comment by pm4rcin** [comment]
@neVERberleRfellerER how did you manage the game to work? I installed G2NR from GOG via Lutris and applied patches like Union (Systempack), but it doesn't want to work. It worked with default install but with patches it starts and then it is stuck, game window isn't drawn. I haven't tried it with DX11 maybe that's the reason? Have you already tried without DX11?

**Comment by neVERberleRfellerER** [comment]
AFAIK union does not work, not even the systempack from union. I use standalone systempack.

**Comment by stefson** [comment]
it does work up to a certain point. How to reproduce:

* Install Gothic2 in win32 prefix (I used Softwarepyramide Gold Version, GOG should work as well)
* Install gothic2_fix-2.6.0.0-rev2.exe into same prefix
* Install G2NoTR-SystemPack-1.8.exe into same prefix
* Install gothic2_playerkit-2.6f.exe into same prefix

* Install dxvk libs via `winetricks dxvk153`
* Go to https://github.com/Kirides/GD3D11/releases/tag/v17.7-dev11 and download Gothic2 related zip file
* unpack the zip file into Gothic2/System folder
* add dll overwrite for ddraw.dll, and ignore warnings

fire up Gothic2 via GothicStarter.exe with `env DXVK_HUD=devinfo,fps,gpuload WINEPREFIX=~/.gothic2-goldedition WINEARCH=win32 wine GothicStarter.exe`

todo: add LAA execution bit, also known as 4GB Patch 

tested with wine-5.22 on gentoo amd64 (multilib)

**Comment by stefson** [comment]
the whole prefix is broken, as soon as `G3D11` is updated beyond dev11: 

```
002c:err:winediag:WS_getaddrinfo Failed to resolve your host name IP
WARNING: Experimental compiler backend enabled. Here be dragons! Incorrect rendering, GPU hangs and/or resets are likely
0108:err:module:import_dll Loading library d3dcompiler_47.dll (which is needed by L"C:\\windows\\system32\\d3d10.dll") failed (error c000035a).
0108:err:module:find_forwarded_export module not found for forward 'd3d10.D3D10CreateStateBlock' used by L"C:\\windows\\system32\\d3d10_1.dll"
0108:err:module:import_dll Loading library d3dcompiler_47.dll (which is needed by L"C:\\windows\\system32\\d3d10.dll") failed (error c000035a).
0108:err:module:find_forwarded_export module not found for forward 'd3d10.D3D10StateBlockMaskEnableAll' used by L"C:\\windows\\system32\\d3d10_1.dll"
0108:err:module:import_dll Loading library D3DCOMPILER_47.dll (which is needed by L"C:\\Program Files\\JoWood\\Gothic II Gold\\System\\DDRAW.dll") failed (error c000035a).
0108:err:module:import_dll Library DDRAW.dll (which is needed by L"C:\\Program Files\\JoWood\\Gothic II Gold\\System\\Gothic2.exe") not found
0108:err:module:LdrInitializeThunk Importing dlls for L"C:\\Program Files\\JoWood\\Gothic II Gold\\System\\Gothic2.exe" failed, status c0000135
```

@Kirides do you maybe know why this happens? I know that you upgraded to dxgi-1.2 in https://github.com/Kirides/GD3D11/commit/8cd96bdea164b6c27f213e61db59099a53f309d7 , but dxvk does support it since 2018.

**Comment by Kirides** [comment]
The error says it all, it tries to load d3dcompiler_47.dll but can not find it.

Maybe you have to add some wine dll shenanigans for it to properly work

**Comment by stefson** [comment]
thanks for your fast answer, do you think this data from Gothic2.exe proves your point? Main difference is more verbose `find_forwarded_export module not found for forward` error


$ env DXVK_HUD=devinfo,fps,gpuload WINEPREFIX=~/.gothic2-goldedition-reboot-mingw32 WINEARCH=win32 wine Gothic2.exe
```
0024:err:module:import_dll Loading library d3dcompiler_47.dll (which is needed by L"C:\\windows\\system32\\d3d10.dll") failed (error c000035a).
0024:err:module:find_forwarded_export module not found for forward 'd3d10.D3D10CreateStateBlock' used by L"C:\\windows\\system32\\d3d10_1.dll"
0024:err:module:import_dll Loading library d3dcompiler_47.dll (which is needed by L"C:\\windows\\system32\\d3d10.dll") failed (error c000035a).
0024:err:module:find_forwarded_export module not found for forward 'd3d10.D3D10StateBlockMaskEnableAll' used by L"C:\\windows\\system32\\d3d10_1.dll"
0024:err:module:import_dll Loading library D3DCOMPILER_47.dll (which is needed by L"C:\\Program Files\\JoWood\\Gothic II Gold\\System\\DDRAW.dll") failed (error c000035a).
0024:err:module:import_dll Library DDRAW.dll (which is needed by L"C:\\Program Files\\JoWood\\Gothic II Gold\\System\\Gothic2.exe") not found
0024:err:module:LdrInitializeThunk Importing dlls for L"C:\\Program Files\\JoWood\\Gothic II Gold\\System\\Gothic2.exe" failed, status c0000135
```

I've been talking to the dxvk author, he confirmed that dxvk has dxgi-1.2 support since 2018. 

It doesn't work to use `winetricks d3dcompiler_47`

**Comment by pm4rcin** [comment]
@Kirides check that reply https://github.com/doitsujin/dxvk/issues/1844#issuecomment-739844878 and decide what to do now.

**Comment by Kirides** [comment]
Just do as he says and remove the existing D3dcompiler_47.dll from the Gothic\System\ directory and try it after installing the dll using winetricks.

The D3DCompiler_47.dll is only released with this renderer because on some systems it was simply not working even though DirectX was installed using the dxweb-install.

**Comment by stefson** [comment]
just for the record: the wine builtin d3dcompiler_47.dll is not working, and to counter more problems down the road it is, as of now, best practice to remove the d3dcompiler_47.dll provided by the GD3D11 archive and use winetricks to do provide your prefix with dll override of d3dcompiler_47.dll 

I do have a question though: can the LAA flag (nicknamed 4GB patch) be set with linux tools?

**Comment by Kirides** [comment]
LAA flag is a windows thing in windows binaries. You should be able to use any tool to set the LAA Bit, even use some windows tool inside that prefix and set it accordingly.

**Comment by stefson** [comment]
Yeah, it's possible to use the tool called `LargeAddressAware Hack` , but I need to boot into my old win7 to use that. I'll report back here, once I found a reasonable way to solve the issue with linux tools.

**Comment by Smokolak** [comment]
Hi, I need some clarifications. I want to get Gothic 1 running with GD3D11.
I installed G1 with Lutris. I got a 32 bit prefix for it. Installed player kit 1.08, 2.08 and system pack 1.8. To this point everything works apart that i need to use BorderlessWindow=1. I'm using built in lutris wine 6.0 rc1 , dxvk 1.7.3L-03f11ba.
In wine configuration I added d3dcompiler_47 and marked as builtin while renamming d3dcompiler_47.dll.old in Gothic/system. Still it doesn't work, what I'm I doing wrong here?

**Comment by stefson** [comment]
why don't you provide your relevant logs? from what you provided, one can only see you're using super bleeding edge packages.

**Comment by neVERberleRfellerER** [comment]
@Smokolak 

> d3dcompiler_47 and marked as builtin

should be set to native (builtin is from wine, native is from Windows). Builtin d3dcompilers are unsupported by dxvk.

**Comment by Smokolak** [comment]
![image](https://user-images.githubusercontent.com/9391918/103279588-60241480-49ce-11eb-9fb3-40641dbdcc0a.png)
![image](https://user-images.githubusercontent.com/9391918/103279655-8d70c280-49ce-11eb-8b9c-ebf4154f5b55.png)
![image](https://user-images.githubusercontent.com/9391918/103279678-9c577500-49ce-11eb-8194-e89248f9ea7a.png)
`

- lutris-wrapper: Gothic
- Running gamemoderun /home/niedzwiedz/.local/share/lutris/runners/wine/lutris-6.0-rc1-x86_64/bin/wine /mnt/b9beef88-d40b-4870-8114-f91037676d46/Games/gothic/drive_c/GOG Games/Gothic/system/GothicStarter.exe
- Initial process has started with pid 341906
- Start monitoring process.
- gamemodeauto: 
- info:  Game: GothicMod.exe
- info:  DXVK: v1.7.3-4-g03f11baf
- info:  Built-in extension providers:
- info:    Win32 WSI
- info:    OpenVR
- warn:  OpenVR: Failed to locate module
- info:  Enabled instance extensions:
- info:    VK_KHR_get_surface_capabilities2
- info:    VK_KHR_surface
- info:    VK_KHR_win32_surface
- warn:  D3D9: VK_FORMAT_D24_UNORM_S8_UINT -> VK_FORMAT_D32_SFLOAT_S8_UINT
- info:  AMD RADV NAVI10 (ACO):
- info:    Driver: 20.2.6
- info:    Vulkan: 1.2.131
- info:    Memory Heap[0]: 
- info:      Size: 7936 MiB
- info:      Flags: 0x1
- info:      Memory Type[0]: Property Flags = 0x1
- info:      Memory Type[4]: Property Flags = 0xc1
- info:    Memory Heap[1]: 
- info:      Size: 8176 MiB
- info:      Flags: 0x0
- info:      Memory Type[1]: Property Flags = 0x6
- info:      Memory Type[3]: Property Flags = 0xe
- info:      Memory Type[5]: Property Flags = 0xc6
- info:      Memory Type[7]: Property Flags = 0xce
- info:    Memory Heap[2]: 
- info:      Size: 256 MiB
- info:      Flags: 0x1
- info:      Memory Type[2]: Property Flags = 0x7
- info:      Memory Type[6]: Property Flags = 0xc7
- info:  Process set as DPI aware
`
Sorry for dumb questions but I'm a noob.
So above is from log named GothicMod_d3d9.log
There are two more logs created in game dir but they are emply GothicMod_d3d11.log and GothicMod_dxgi.log when game is started.
I observed that when I add BorderlessWindow=1 game starts but there aren't any GD3D11 effects applied in window mode.
When I start game with BorderlessWindow=0 I hear the sound (intro/menu), process is running but nothing is displayed. Log is stuck at Process set as DPI aware and nothing new is added there.
[sp.log](https://github.com/Kirides/GD3D11/files/5750549/sp.log)

**Comment by stefson** [comment]
the GD3D11 renderer ships its own ddraw.dll, I think this one you have to set to override the one provided by wine prefix: `ddraw (Native,Builtin)` this is at least true for Gothic2

**Comment by Smokolak** [comment]
> the GD3D11 renderer ships its own ddraw.dll, I think this one you have to set to override over the one provided by wine prefix: `ddraw (Native,Builtin)` this is at least true for Gothic2
Awesome! It works! Thanks! But I had to revert d3dcompiler_47.dll to game dir back.

**Comment by stefson** [comment]
Correct, builtin d3dcompiler_47.dll implementation is somehow incomplete or broken. I may open a bug on wine bug tracker for that. Until it is fixed, you should use the provided d3dcompiler_47.dll and not remove or rename it. It might be possible to use winetricks to provide the d3dcompiler_47.dll override for you. 

You are using super bleeding edge software, breakages can and will occure. GD3D11 itself is somehow alpha or pre alpha, and while it's a good thing you're out there and testing the snapshots of wine and dxvk for that matter, you should act on crashes and report them. But only if you're confident they don't happen with the latest stable, making them a regression. 

In any case, for Gothic2 the frequent crashes can be solved by adding larg adress hack flag to Gothic2.exe. Just scroll up a bit, I talked about it in this thread before.

**Comment by pm4rcin** [comment]
I managed to install 17.7-dev16 on G2NR. It's probably fine to go with SystemPack 1.8 but recently Union (SystemPack 1.9) added support for wine so you can use plugins e.g. zGamepad plugin which works fine even on Steam Controller. Steps to reproduce:
- Install game on **fresh prefix** via Lutris, Steam etc.
- Add **vcrun2008, vcrun2010, dotnet40** with winetricks or protontricks respectively needed for ResourceManager and Union.
- Download **Resource Manager** and run it in Gothic prefix: https://worldofplayers.ru/threads/41415/
- Install **patch for 64-bit** aka **4GB patch** and **Union** (currently tested newest 1.0j).
- Download zip file for your Gothic version from [Releases](https://github.com/Kirides/GD3D11/releases) page.
- Unpack zip content to **Gothic\System** folder.
- **Override ddraw** in prefix (for Lutris it's in wine configuration -> Libraries -> Type ddraw in the text field, click "Add" and confirm).
- Set GothicStarter.exe as default executable program.
- You're good to go. Mangohud also works. :)

**Comment by pm4rcin** [comment]
> Awesome! It works! Thanks! But I had to revert d3dcompiler_47.dll to game dir back.
> 
> Unfortunately I'm getting some errors and it crash after some time:
> 
>     * lutris-wrapper: Gothic
>       Running gamemoderun /home/niedzwiedz/.local/share/lutris/runners/wine/lutris-6.0-rc1-x86_64/bin/wine /mnt/b9beef88-d40b-4870-8114-f91037676d46/Games/gothic/drive_c/GOG Games/Gothic/system/GothicStarter.exe
>       Initial process has started with pid 349522
>       Start monitoring process.
>       gamemodeauto:
>       esync: up and running.
>       info:  Game: GothicMod.exe
>       info:  DXVK: v1.7.3-4-g03f11baf
>       info:  Built-in extension providers:
>       info:    Win32 WSI
>       info:    OpenVR
>       warn:  OpenVR: Failed to locate module
>       info:  Enabled instance extensions:
>       info:    VK_KHR_get_surface_capabilities2
>       info:    VK_KHR_surface
>       info:    VK_KHR_win32_surface
>       warn:  D3D9: VK_FORMAT_D24_UNORM_S8_UINT -> VK_FORMAT_D32_SFLOAT_S8_UINT
>       info:  AMD RADV NAVI10 (ACO):
>       info:    Driver: 20.2.6
>       info:    Vulkan: 1.2.131
>       info:    Memory Heap[0]:
>       info:      Size: 7936 MiB
>       info:      Flags: 0x1
>       info:      Memory Type[0]: Property Flags = 0x1
>       info:      Memory Type[4]: Property Flags = 0xc1
>       info:    Memory Heap[1]:
>       info:      Size: 8176 MiB
>       info:      Flags: 0x0
>       info:      Memory Type[1]: Property Flags = 0x6
>       info:      Memory Type[3]: Property Flags = 0xe
>       info:      Memory Type[5]: Property Flags = 0xc6
>       info:      Memory Type[7]: Property Flags = 0xce
>       info:    Memory Heap[2]:
>       info:      Size: 256 MiB
>       info:      Flags: 0x1
>       info:      Memory Type[2]: Property Flags = 0x7
>       info:      Memory Type[6]: Property Flags = 0xc7
>       info:  Process set as DPI aware
>       info:  Game: GothicMod.exe
>       info:  DXVK: v1.7.3-4-g03f11baf
>       info:  Built-in extension providers:
>       info:    Win32 WSI
>       info:    OpenVR
>       warn:  OpenVR: Failed to locate module
>       info:  Enabled instance extensions:
>       info:    VK_KHR_get_surface_capabilities2
>       info:    VK_KHR_surface
>       info:    VK_KHR_win32_surface
>       info:  AMD RADV NAVI10 (ACO):
>       info:    Driver: 20.2.6
>       info:    Vulkan: 1.2.131
>       info:    Memory Heap[0]:
>       info:      Size: 7936 MiB
>       info:      Flags: 0x1
>       info:      Memory Type[0]: Property Flags = 0x1
>       info:      Memory Type[4]: Property Flags = 0xc1
>       info:    Memory Heap[1]:
>       info:      Size: 8176 MiB
>       info:      Flags: 0x0
>       info:      Memory Type[1]: Property Flags = 0x6
>       info:      Memory Type[3]: Property Flags = 0xe
>       info:      Memory Type[5]: Property Flags = 0xc6
>       info:      Memory Type[7]: Property Flags = 0xce
>       info:    Memory Heap[2]:
>       info:      Size: 256 MiB
>       info:      Flags: 0x1
>       info:      Memory Type[2]: Property Flags = 0x7
>       info:      Memory Type[6]: Property Flags = 0xc7
>       info:  D3D11CoreCreateDevice: Probing D3D_FEATURE_LEVEL_11_0
>       info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_11_0
>       info:  Device properties:
>       info:    Device name:     : AMD RADV NAVI10 (ACO)
>       info:    Driver version   : 20.2.6
>       info:  Enabled device extensions:
>       info:    VK_AMD_memory_overallocation_behavior
>       info:    VK_AMD_shader_fragment_mask
>       info:    VK_EXT_4444_formats
>       info:    VK_EXT_custom_border_color
>       info:    VK_EXT_depth_clip_enable
>       info:    VK_EXT_extended_dynamic_state
>       info:    VK_EXT_full_screen_exclusive
>       info:    VK_EXT_host_query_reset
>       info:    VK_EXT_memory_priority
>       info:    VK_EXT_robustness2
>       info:    VK_EXT_shader_demote_to_helper_invocation
>       info:    VK_EXT_shader_stencil_export
>       info:    VK_EXT_shader_viewport_index_layer
>       info:    VK_EXT_transform_feedback
>       info:    VK_EXT_vertex_attribute_divisor
>       info:    VK_KHR_create_renderpass2
>       info:    VK_KHR_depth_stencil_resolve
>       info:    VK_KHR_draw_indirect_count
>       info:    VK_KHR_driver_properties
>       info:    VK_KHR_image_format_list
>       info:    VK_KHR_sampler_mirror_clamp_to_edge
>       info:    VK_KHR_swapchain
>       info:  Device features:
>       info:    robustBufferAccess                     : 1
>       info:    fullDrawIndexUint32                    : 1
>       info:    imageCubeArray                         : 1
>       info:    independentBlend                       : 1
>       info:    geometryShader                         : 1
>       info:    tessellationShader                     : 1
>       info:    sampleRateShading                      : 1
>       info:    dualSrcBlend                           : 1
>       info:    logicOp                                : 1
>       info:    multiDrawIndirect                      : 1
>       info:    drawIndirectFirstInstance              : 1
>       info:    depthClamp                             : 1
>       info:    depthBiasClamp                         : 1
>       info:    fillModeNonSolid                       : 1
>       info:    depthBounds                            : 1
>       info:    multiViewport                          : 1
>       info:    samplerAnisotropy                      : 1
>       info:    textureCompressionBC                   : 1
>       info:    occlusionQueryPrecise                  : 1
>       info:    pipelineStatisticsQuery                : 1
>       info:    vertexPipelineStoresAndAtomics         : 0
>       info:    fragmentStoresAndAtomics               : 1
>       info:    shaderImageGatherExtended              : 1
>       info:    shaderStorageImageExtendedFormats      : 1
>       info:    shaderStorageImageReadWithoutFormat    : 1
>       info:    shaderStorageImageWriteWithoutFormat   : 1
>       info:    shaderClipDistance                     : 1
>       info:    shaderCullDistance                     : 1
>       info:    shaderFloat64                          : 1
>       info:    shaderInt64                            : 1
>       info:    variableMultisampleRate                : 1
>       info:  VK_EXT_4444_formats
>       info:    formatA4R4G4B4                         : 1
>       info:    formatA4B4G4R4                         : 1
>       info:  VK_EXT_custom_border_color
>       info:    customBorderColors                     : 1
>       info:    customBorderColorWithoutFormat         : 1
>       info:  VK_EXT_depth_clip_enable
>       info:    depthClipEnable                        : 1
>       info:  VK_EXT_extended_dynamic_state
>       info:    extendedDynamicState                   : 1
>       info:  VK_EXT_host_query_reset
>       info:    hostQueryReset                         : 1
>       info:  VK_EXT_memory_priority
>       info:    memoryPriority                         : 1
>       info:  VK_EXT_robustness2
>       info:    robustBufferAccess2                    : 1
>       info:    robustImageAccess2                     : 1
>       info:    nullDescriptor                         : 1
>       info:  VK_EXT_shader_demote_to_helper_invocation
>       info:    shaderDemoteToHelperInvocation         : 1
>       info:  VK_EXT_transform_feedback
>       info:    transformFeedback                      : 1
>       info:    geometryStreams                        : 1
>       info:  VK_EXT_vertex_attribute_divisor
>       info:    vertexAttributeInstanceRateDivisor     : 1
>       info:    vertexAttributeInstanceRateZeroDivisor : 1
>       info:  Queue families:
>       info:    Graphics : 0
>       info:    Transfer : 1
>       info:  DXVK: Read 128 valid state cache entries
>       info:  DXVK: Using 7 compiler threads
>       warn:  DXGI: VK_FORMAT_D24_UNORM_S8_UINT -> VK_FORMAT_D32_SFLOAT_S8_UINT
>       err:   D3D11RasterizerState: Antialiased lines not supported
>       err:   D3D11RasterizerState: Antialiased lines not supported
>       warn:  winevulkan detected, disabling exclusive fullscreen support
>       info:  Presenter: Actual swap chain properties:
>       info:    Format:       VK_FORMAT_B8G8R8A8_UNORM
>       info:    Present mode: VK_PRESENT_MODE_IMMEDIATE_KHR
>       info:    Buffer size:  2560x1080
>       info:    Image count:  3
>       info:    Exclusive FS: 2
>       err:   D3D11RasterizerState: Antialiased lines not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       warn:  D3D11Device::CreateShaderModule: Class linkage not supported
>       info:  Presenter: Actual swap chain properties:
>       info:    Format:       VK_FORMAT_B8G8R8A8_UNORM
>       info:    Present mode: VK_PRESENT_MODE_FIFO_KHR
>       info:    Buffer size:  2560x1080
>       info:    Image count:  4
>       info:    Exclusive FS: 2
>       err:   D3D11RasterizerState: Antialiased lines not supported
>       err:   D3D11RasterizerState: Antialiased lines not supported
>       err:   D3D11RasterizerState: Antialiased lines not supported
>       err:   D3D11RasterizerState: Antialiased lines not supported
>       err:   D3D11RasterizerState: Antialiased lines not supported
>       err:   D3D11RasterizerState: Antialiased lines not supported
>       err:   D3D11RasterizerState: Antialiased lines not supported

I have the same problem with crashes on Nvidia. When I'm on **Jarkendar** and use marvin+f8 to fly after a short time the game crashes. But it's probably not DX11's problem because when I moved the files and let the original renderer do a job the same thing happened... Also I started new game and moved fast to the teleport to Jarkendar with f8 and a short time after teleporting the game has freezed.

**Comment by pm4rcin** [comment]
If you have any problems or questions post it there. I'll try my best to resolve them. :)
So to successfully get the game working on Linux (I used Lutris) you have to:

1. Install the game on prefix (I used gog version with automatic install from Lutris)
2. Add vcrun2008, vcrun2010, dotnet40 with winetricks (you can replace dotnet40 with globally installed wine-mono)
3. Download [Resource Manager](https://worldofplayers.ru/resources/136/updates) and place it in the prefix folder
4. In Resource Manager on Union tab install Patch for 64-bit systems AND Union (minimum 1.0j)
5. Make sure that directmusic was added with winetricks
6. Make sure that dsound is set to built-in and override ddraw to native(or whatever the name is (can't be built-in, it has to be the second option) in winecfg
7. [Download](https://github.com/Kirides/GD3D11/releases) and unpack DX11 files
8. (Optional) Download normalmaps from [here](https://www.worldofgothic.de/dl/download_590.htm)
9. (Optional) Add plugins with Resource Manager (I've tested zGamePad and it works flawlessly even with with SteamController)

Specs:
CPU: Ryzen 5 4600H
GPU: Geforce GTX 1650
RAM: 16GB 3200MHz
Dist: Arch 5.11.16
Wine: System(vanilla 6.7)

**Comment by stefson** [comment]
@pm4rcin were you able to add the laa hack with linux tools?

**Comment by bodzioslav** [comment]
@stefson 
Thanks, your guide works fine! Instead of dxvk173 I went with dxvk181 and I also installed directmusicthat way. I also used 4GB path.

**Comment by stefson** [comment]
sure, you can replace with latest dxvk and also with latest GD3D11 release tag, glad it still works. Did you use any linux tools for the 4gb patch?

**Comment by bodzioslav** [comment]
@stefson on, I just did  `WINEPREFIX="~/.wine/prefix/gothic2nk" WINEARCH="win32" wine ~/installers/LaaHack.exe` and then I checked out `Gothic2.exe` and pressed `exit`. After doing so the game stopped crashing when I was watching in game loading movies.

**Comment by pm4rcin** [comment]
> @pm4rcin were you able to add the laa hack with linux tools?

The patch for 64-bit systems includes 4GB Patch in it so I haven't had to do that. It's in description of that patch. ;)

**Comment by bodzioslav** [comment]
@pm4rcin The performance is good and in general the game works for me (G2: NoTR) but sometimes my screen goes blank and the only way to fix this is to kill the Gothic2.exe process and start the game again. 
Do you have any idea what can be the cause of the issue?

**Comment by pm4rcin** [comment]
> @pm4rcin The performance is good and in general the game works for me (G2: NoTR) but sometimes my screen goes blank and the only way to fix this is to kill the Gothic2.exe process and start the game again.
> Do you have any idea what can be the cause of the issue?

1. Did you install 4GB patch?
2. Prefix 32 or 64-bit?
3. Wine version?
4. Kernel version, distro?
5. SystemPack/Union and which version?
6.  DX11 installed and which version?
7. If yes did you check with original renderer?

**Comment by bodzioslav** [comment]
> > @pm4rcin The performance is good and in general the game works for me (G2: NoTR) but sometimes my screen goes blank and the only way to fix this is to kill the Gothic2.exe process and start the game again.
> > Do you have any idea what can be the cause of the issue?
> 
>     1. Did you install 4GB patch?
Yes, otherwise the game crashed due to `bad alloc`.
>     2. Prefix 32 or 64-bit?
32bit
>     3. Wine version?
6.1
>     4. Kernel version, distro?
Linux pc 5.4.0-65-generic #73-Ubuntu SMP Mon Jan 18 17:25:17 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

>     5. SystemPack/Union and which version?
System Pack 1.8
>     6. DX11 installed and which version?
dxvk 1.8.1 with GD3D11 17.7-dev11-fix2
>     7. If yes did you check with original renderer?
With original renderer the game menu is laggy so I haven't tried it yet. But I do play Gothic 1 with original renderer because it crashes with ddraw override.

**Comment by stefson** [comment]
first thing you should try is to upgrade the gd3d11 renderer with the latest snapshot. also posting your gpu and which drivers you're using can be helpfull.

**Comment by bodzioslav** [comment]
Sure. I will also update my Ubuntu to the recent version available.

My specs:
AMD Ryzen 5 2600
Radeon RX580 8GB with `amdgpu` driver.
16GB DDR4 3200MHz (dual channel)

With the newest GD3D11 Gothic 1 still crashes due to `bad alloc`. I will try playing Gothic2 NoTR and see what happens.

**Comment by pm4rcin** [comment]
@bodzioslav using 32-bit prefix with 4GB patch is risky in my opinion. You could try these steps since I tested them many times and it always worked for me. But I have Nvidia graphics so AMD could be the issue on your side.

**Comment by bodzioslav** [comment]
So should I try using 64bit prefix instead?

**Comment by pm4rcin** [comment]
Sorry I meant https://github.com/Kirides/GD3D11/issues/14#issuecomment-827858549 since it's reproducible every time.

**Comment by bodzioslav** [comment]
I have updated GD3D11 and also the whole system. Now I'm running wine 6.7 and Linux kernel `5.11.0-16-generic` I have managed to play for 2h 30m without any issues so for now I assume that in case of G2NoTR it helped. 

In case of Gothic1 I haven't succeeded yet. It still crashes with GD3D11 enabled (dddraw override). I also tried to reinstall G1 inside of a wine64 prefix and the issue is still there. Unfortunately 4GB patch doesn't fix G1.
